File: sdk/runanywhere-swift/Sources/RunAnywhere/Core/Models/Framework/LLMFramework.swift
import Foundation

/// Supported LLM frameworks
public enum LLMFramework: String, CaseIterable, Codable {
    case coreML = "CoreML"
    case tensorFlowLite = "TFLite"
    case mlx = "MLX"
    case swiftTransformers = "SwiftTransformers"
    case onnx = "ONNX"
    case execuTorch = "ExecuTorch"
    case llamaCpp = "LlamaCpp"
    case foundationModels = "FoundationModels"
    case picoLLM = "PicoLLM"
    case mlc = "MLC"
    case mediaPipe = "MediaPipe"
    case whisperKit = "WhisperKit"
    case openAIWhisper = "OpenAIWhisper"

    /// Human-readable display name for the framework
    public var displayName: String {
        switch self {
        case .coreML: return "Core ML"
        case .tensorFlowLite: return "TensorFlow Lite"
        case .mlx: return "MLX"
        case .swiftTransformers: return "Swift Transformers"
        case .onnx: return "ONNX Runtime"
        case .execuTorch: return "ExecuTorch"
        case .llamaCpp: return "llama.cpp"
        case .foundationModels: return "Foundation Models"
        case .picoLLM: return "Pico LLM"
        case .mlc: return "MLC"
        case .mediaPipe: return "MediaPipe"
        case .whisperKit: return "WhisperKit"
        case .openAIWhisper: return "OpenAI Whisper"
        }
    }
}

File: sdk/runanywhere-android/src/main/java/com/runanywhere/sdk/models/LLMFramework.kt
package com.runanywhere.sdk.models

/**
 * Supported LLM frameworks
 */
enum class LLMFramework(val displayName: String) {
    TENSORFLOW_LITE("TensorFlow Lite"),
    ONNX("ONNX Runtime"),
    EXECUTORCH("ExecuTorch"),
    LLAMACPP("llama.cpp"),
    FOUNDATION_MODELS("Foundation Models"),
    PICOLLM("Pico LLM"),
    MLC("MLC"),
    MEDIAPIPE("MediaPipe"),
    NCNN("NCNN"),
    OPENVINO("OpenVINO"),
    TFLITE_GPU("TensorFlow Lite GPU"),
    TFLITE_NNAPI("TensorFlow Lite NNAPI");
    
    companion object {
        fun fromString(value: String): LLMFramework? {
            return values().find { it.name.equals(value, ignoreCase = true) }
        }
    }
} 